From 100a77f4d19ea3195c1c1f2faf459b5c278e6999 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Tue, 11 Feb 2025 17:40:46 +0800
Subject: [PATCH 7/9] overlayfs:dont make DCACHE_OP_{HASH,COMPARE} weird

---
 fs/overlayfs/util.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/fs/overlayfs/util.c b/fs/overlayfs/util.c
index b103ce250..58807ec08 100644
--- a/fs/overlayfs/util.c
+++ b/fs/overlayfs/util.c
@@ -143,9 +143,7 @@ bool ovl_dentry_weird(struct dentry *dentry)
 		return true;
 
 	return dentry->d_flags & (DCACHE_NEED_AUTOMOUNT |
-				  DCACHE_MANAGE_TRANSIT |
-				  DCACHE_OP_HASH |
-				  DCACHE_OP_COMPARE);
+				  DCACHE_MANAGE_TRANSIT);
 }
 
 enum ovl_path_type ovl_path_type(struct dentry *dentry)
-- 
2.48.1

