From 675c40a50678425e9c0f7a4997273daff4890c23 Mon Sep 17 00:00:00 2001
From: flakeforever <113103193+flakeforever@users.noreply.github.com>
Date: Sat, 18 May 2024 09:58:12 +0800
Subject: [PATCH] build: Disable build kernel module

---
 build/tasks/kernel.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/tasks/kernel.mk b/build/tasks/kernel.mk
index 0c2e179e..03c8ded7 100644
--- a/build/tasks/kernel.mk
+++ b/build/tasks/kernel.mk
@@ -464,7 +464,7 @@ $(TARGET_PREBUILT_INT_KERNEL): $(KERNEL_CONFIG) $(DEPMOD) $(DTC)
 			echo "Building DTBs"; \
 			$(call make-kernel-target,dtbs); \
 		fi
-	$(hide) if grep -q '=m' $(KERNEL_CONFIG); then \
+	$(hide) if grep -q '=m' $(KERNEL_CONFIG) && false; then \
 			echo "Building Kernel Modules"; \
 			$(call make-kernel-target,modules) || exit "$$?"; \
 			echo "Installing Kernel Modules"; \
-- 
2.34.1

