From f39ea568da9d60ec7fc6698e18b03ee2751d0cb1 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Fri, 29 Nov 2024 14:53:56 +0800
Subject: [PATCH] fix missing include

---
 hal/audio_extn/Android.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hal/audio_extn/Android.mk b/hal/audio_extn/Android.mk
index b439d65..59d11f4 100644
--- a/hal/audio_extn/Android.mk
+++ b/hal/audio_extn/Android.mk
@@ -46,6 +46,7 @@ LOCAL_C_INCLUDES := \
     $(LOCAL_PATH)/inc \
     external/expat/lib \
     system/media/audio_utils/include \
+    hardware/qcom-caf/sm8450/audio/pal \
     $(call include-path-for, audio-route) \
 
 LOCAL_HEADER_LIBRARIES += libhardware_headers
-- 
2.39.2

