From 9331cdee46b40c3d16ed18535618612a4e7e40af Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Fri, 29 Nov 2024 15:27:28 +0800
Subject: [PATCH] do not mount ion and mali0, mount dma_heap

---
 lxc/template.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/lxc/template.sh b/lxc/template.sh
index 8e6b2cb..96a2c1f 100755
--- a/lxc/template.sh
+++ b/lxc/template.sh
@@ -416,11 +416,11 @@ lxc.mount.entry = /dev/hwbinder dev/hwbinder none bind,optional,rw,create=file
 lxc.mount.entry = /dev/vndbinder dev/vndbinder none bind,optional,rw,create=file
 lxc.mount.entry = /dev/pmsg0 dev/pmsg0 none bind,optional,rw,create=file
 lxc.mount.entry = /dev/ashmem dev/ashmem none bind,optional,rw,create=file
-lxc.mount.entry = /dev/ion dev/ion none bind,optional,create=file
+#lxc.mount.entry = /dev/ion dev/ion none bind,optional,create=file
 lxc.mount.entry = /dev/input dev/input none rbind,optional,rw,create=dir
 lxc.mount.entry = selinuxfs sys/fs/selinux selinuxfs optional 0 0
 lxc.mount.entry = /dev/binderfs dev/binderfs bind bind,create=dir,optional 0 0
-
+lxc.mount.entry = /dev/dma_heap dev/dma_heap none rbind,optional,rw,create=dir
 # Hack, bind mount patched libc inside container to avoid errors
 lxc.mount.entry = /system_ext/usr/share/lindroid/libc.so apex/com.android.runtime/lib64/bionic/libc.so none bind,optional,ro,create=file
 
@@ -429,7 +429,7 @@ lxc.mount.entry = /data/lindroid/mnt lindroid bind bind,create=dir,optional 0 0
 
 # GPU devices
 lxc.mount.entry = /dev/kgsl-3d0 dev/kgsl-3d0 none bind,optional,create=file
-lxc.mount.entry = /dev/mali0 dev/mali0 none bind,optional,create=file
+#lxc.mount.entry = /dev/mali0 dev/mali0 none bind,optional,create=file
 # -- static Lindroid configuration ends here --
 EOF
 }
-- 
2.39.2

