From a3d6961dc4f3f7b42907c099b394aef3c882ea37 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Mon, 24 Feb 2025 22:05:17 +0800
Subject: [PATCH] do not mount kgsl-3d0 because xaga is a mtk device

---
 lxc/template.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lxc/template.sh b/lxc/template.sh
index 8915df4..a2bfaa9 100755
--- a/lxc/template.sh
+++ b/lxc/template.sh
@@ -416,7 +416,7 @@ lxc.mount.entry = /dev/hwbinder dev/hwbinder none bind,optional,rw,create=file
 lxc.mount.entry = /dev/vndbinder dev/vndbinder none bind,optional,rw,create=file
 lxc.mount.entry = /dev/pmsg0 dev/pmsg0 none bind,optional,rw,create=file
 lxc.mount.entry = /dev/ashmem dev/ashmem none bind,optional,rw,create=file
-lxc.mount.entry = /dev/ion dev/ion none bind,optional,create=file
+lxc.mount.entry = /dev/dma_heap dev/dma_heap none bind,optional,rw,create=dir
 lxc.mount.entry = /dev/input dev/input none rbind,optional,rw,create=dir
 lxc.mount.entry = selinuxfs sys/fs/selinux selinuxfs optional 0 0
 lxc.mount.entry = /dev/binderfs dev/binderfs bind bind,create=dir,optional 0 0
@@ -429,7 +429,7 @@ lxc.mount.entry = /data/lindroid/mnt lindroid bind bind,create=dir,optional 0 0
 
 # GPU devices
 lxc.mount.entry = /dev/dri dev/dri none bind,optional,rw,create=dir
-lxc.mount.entry = /dev/kgsl-3d0 dev/kgsl-3d0 none bind,optional,create=file
+#lxc.mount.entry = /dev/kgsl-3d0 dev/kgsl-3d0 none bind,optional,create=file
 lxc.mount.entry = /dev/mali0 dev/mali0 none bind,optional,create=file
 # -- static Lindroid configuration ends here --
 EOF
-- 
2.50.1

