From effc593e97e31a94bd7aef70024063951994e942 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Wed, 4 Dec 2024 21:56:06 +0800
Subject: [PATCH] Add flags for risingos

---
 lineage_mondrian.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lineage_mondrian.mk b/lineage_mondrian.mk
index f46526b..8c1a7a0 100644
--- a/lineage_mondrian.mk
+++ b/lineage_mondrian.mk
@@ -28,4 +28,8 @@ PRODUCT_BUILD_PROP_OVERRIDES += \
 TARGET_BOOT_ANIMATION_RES := 1440
 
 WITH_GMS := true
+TARGET_CORE_GMS_EXTRAS := true
 TARGET_HAS_UDFPS := true
+TARGET_ENABLE_BLUR := true
+PRODUCT_NO_CAMERA := false
+TARGET_INCLUDE_GOOGLE_DIALER := true
-- 
2.39.2

