From 89df026f7b90c6de21c1ce5258a8231bf72be73d Mon Sep 17 00:00:00 2001
From: HeCheng Yu <kde-yyds@qq.com>
Date: Fri, 10 Jan 2025 22:44:49 +0800
Subject: [PATCH] Add Chipset/Maintainer properties

---
 device.mk | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/device.mk b/device.mk
index 153ef3c..996f98e 100644
--- a/device.mk
+++ b/device.mk
@@ -663,3 +663,8 @@ PRODUCT_PACKAGES += \
 # XiaomiParts
 PRODUCT_PACKAGES += \
     XiaomiParts
+
+# RisingOS
+PRODUCT_BUILD_PROP_OVERRIDES += \
+    RisingChipset="Snapdragon 8+ Gen1" \
+    RisingMaintainer="kde-yyds"
-- 
2.47.0

