From d57820e22125b80940953f623bfc740235d5d8b7 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Wed, 4 Dec 2024 20:26:05 +0800
Subject: [PATCH] inherit lindroid

---
 device.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/device.mk b/device.mk
index 16b2d2d..ff4335c 100644
--- a/device.mk
+++ b/device.mk
@@ -26,6 +26,9 @@ $(call inherit-product, frameworks/native/build/phone-xhdpi-6144-dalvik-heap.mk)
 # Inherit from the proprietary version
 $(call inherit-product, vendor/xiaomi/mondrian/mondrian-vendor.mk)
 
+# Inherit lindroid
+$(call inherit-product, vendor/lindroid/lindroid.mk)
+
 # Platform
 TARGET_BOARD_PLATFORM := taro
 
-- 
2.39.2

