From 2a0db4e8338a8473ef82283193a08c64c042b333 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Sat, 8 Mar 2025 11:30:45 +0800
Subject: [PATCH] mt6895: enable 32-bit

---
 BoardConfigCommon.mk | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index e1e761c..4c0ec1f 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -33,6 +33,13 @@ TARGET_CPU_ABI := arm64-v8a
 TARGET_CPU_ABI2 :=
 TARGET_CPU_VARIANT := cortex-a76
 
+TARGET_2ND_ARCH := arm
+TARGET_2ND_ARCH_VARIANT := armv8-2a
+TARGET_2ND_CPU_ABI := armeabi-v7a
+TARGET_2ND_CPU_ABI2 := armeabi
+TARGET_2ND_CPU_VARIANT := generic
+TARGET_2ND_CPU_VARIANT_RUNTIME := cortex-a75
+
 # Boot Image
 BOARD_KERNEL_BASE         := 0x3fff8000
 BOARD_KERNEL_OFFSET       := 0x00008000
-- 
2.48.1

