From e40e9bcd53ea7ef4183d4fefa5ecf60812ee8fd2 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Wed, 19 Feb 2025 22:15:09 +0800
Subject: [PATCH 2/3] Add gms flags for risingOS

---
 lineage_xaga.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lineage_xaga.mk b/lineage_xaga.mk
index 5f38f72..eb24358 100644
--- a/lineage_xaga.mk
+++ b/lineage_xaga.mk
@@ -22,6 +22,10 @@ TARGET_BOOT_ANIMATION_RES := 1080
 # ROM Flags
 TARGET_DISABLE_EPPE := true
 WITH_GMS := true
+TARGET_CORE_GMS_EXTRAS := true
+TARGET_ENABLE_BLUR := true
+PRODUCT_NO_CAMERA := false
+TARGET_INCLUDE_GOOGLE_DIALER := true
 
 PRODUCT_BRAND := Redmi
 PRODUCT_DEVICE := xaga
-- 
2.39.2

