From 39fe0edafb22344312b18059c769f8d32bf9e88b Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Wed, 19 Feb 2025 22:16:07 +0800
Subject: [PATCH 3/3] Disable default frame rate limit for games

---
 vendor.prop | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/vendor.prop b/vendor.prop
index 4e020e6..fbb0705 100644
--- a/vendor.prop
+++ b/vendor.prop
@@ -35,3 +35,6 @@ persist.vendor.fingerprint.sensor_location=1080|960|200|local:462703942230018764
 
 # FM Radio
 ro.vendor.fm.platform=connac2x
+
+# Disable default frame rate limit for games
+debug.graphics.game_default_frame_rate.disabled=1
-- 
2.39.2

