From e2c179d024f0b88f30d0445ea8b69a1f3165cee7 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Thu, 28 Nov 2024 22:13:20 +0800
Subject: [PATCH]  inputflinger: allow disabling input devices via idc

---
 services/inputflinger/reader/EventHub.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/services/inputflinger/reader/EventHub.cpp b/services/inputflinger/reader/EventHub.cpp
index fe70a51..ca0428b 100644
--- a/services/inputflinger/reader/EventHub.cpp
+++ b/services/inputflinger/reader/EventHub.cpp
@@ -2355,6 +2355,13 @@ void EventHub::openDeviceLocked(const std::string& devicePath) {
     // Load the configuration file for the device.
     device->loadConfigurationLocked();
 
+    // Disable device if device config property set
+    if (device->configuration &&
+        device->configuration->getBool("device.disabled")) {
+        device->disable();
+        ALOGV("Disabling device with id %d\n", device->id);
+    }
+
     // Figure out the kinds of events the device reports.
     device->readDeviceBitMask(EVIOCGBIT(EV_KEY, 0), device->keyBitmask);
     device->readDeviceBitMask(EVIOCGBIT(EV_ABS, 0), device->absBitmask);
-- 
2.39.2

