From 5eaa45699444255454acd3d48d88eab94ee9f0ab Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Fri, 29 Nov 2024 13:57:50 +0800
Subject: [PATCH] fix hostapd compilation error

---
 src/drivers/driver_nl80211_event.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/drivers/driver_nl80211_event.c b/src/drivers/driver_nl80211_event.c
index 32786cb..b6a7c85 100644
--- a/src/drivers/driver_nl80211_event.c
+++ b/src/drivers/driver_nl80211_event.c
@@ -2780,7 +2780,7 @@ static void qca_nl80211_key_mgmt_auth(struct wpa_driver_nl80211_data *drv,
 			   tb[QCA_WLAN_VENDOR_ATTR_ROAM_AUTH_PMKID],
 			   tb[QCA_WLAN_VENDOR_ATTR_ROAM_AUTH_MLO_LINKS]);
 
-#ifdef ANDROID
+#if 0
 #ifdef ANDROID_LIB_EVENT
 	wpa_driver_nl80211_driver_event(
 		drv, OUI_QCA, QCA_NL80211_VENDOR_SUBCMD_KEY_MGMT_ROAM_AUTH,
-- 
2.39.2

