From 81a7ff6cd2b1f8a67eba312f4c6926db332d8834 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Thu, 28 Nov 2024 22:32:00 +0800
Subject: [PATCH] disable selinux

---
 init/selinux.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/init/selinux.cpp b/init/selinux.cpp
index c2d9b8d..c96834b 100644
--- a/init/selinux.cpp
+++ b/init/selinux.cpp
@@ -106,6 +106,7 @@ EnforcingStatus StatusFromProperty() {
 }
 
 bool IsEnforcing() {
+    return false;
     if (ALLOW_PERMISSIVE_SELINUX) {
         return StatusFromProperty() == SELINUX_ENFORCING;
     }
-- 
2.39.2

