From 8efa24ab5330b9104f89ab9931e43bcd92e7be60 Mon Sep 17 00:00:00 2001
From: flakeforever <113103193+flakeforever@users.noreply.github.com>
Date: Sat, 23 Nov 2024 18:47:19 +0800
Subject: [PATCH] AutoBrightness: Add support for rear light sensor in
 auto-brightness

---
 .../AutomaticBrightnessController.java        | 43 ++++++++++++++++++-
 1 file changed, 41 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/display/AutomaticBrightnessController.java b/services/core/java/com/android/server/display/AutomaticBrightnessController.java
index c7d9a46f6809..86848ccb7ebd 100644
--- a/services/core/java/com/android/server/display/AutomaticBrightnessController.java
+++ b/services/core/java/com/android/server/display/AutomaticBrightnessController.java
@@ -56,6 +56,7 @@ import com.android.server.EventLogTags;
 import com.android.server.display.brightness.BrightnessEvent;
 import com.android.server.display.config.HysteresisLevels;
 import com.android.server.display.feature.DisplayManagerFlags;
+import com.android.server.display.utils.SensorUtils;
 
 import java.io.PrintWriter;
 import java.lang.annotation.Retention;
@@ -115,6 +116,9 @@ public class AutomaticBrightnessController {
     // The light sensor, or null if not available or needed.
     private final Sensor mLightSensor;
 
+    // The light sensor, or null if not available or needed.
+    private final Sensor mBackLightSensor;
+
     // The mapper to translate ambient lux to screen brightness in the range [0, 1.0].
     @NonNull
     private BrightnessMappingStrategy mCurrentBrightnessMapper;
@@ -281,6 +285,9 @@ public class AutomaticBrightnessController {
 
     private boolean mAutoBrightnessOneShot;
 
+    private float mLux = 0.0f;
+    private float mBackLux = 0.0f;
+
     AutomaticBrightnessController(Callbacks callbacks, Looper looper,
             SensorManager sensorManager, Sensor lightSensor,
             SparseArray<BrightnessMappingStrategy> brightnessMappingStrategyMap,
@@ -358,6 +365,7 @@ public class AutomaticBrightnessController {
 
         if (!DEBUG_PRETEND_LIGHT_SENSOR_ABSENT) {
             mLightSensor = lightSensor;
+            mBackLightSensor = getBackLightSensor();
         }
 
         mActivityTaskManager = ActivityTaskManager.getService();
@@ -520,8 +528,11 @@ public class AutomaticBrightnessController {
         if (mAutoBrightnessOneShot && !autoBrightnessOneShot) {
             mSensorManager.registerListener(mLightSensorListener, mLightSensor,
                     mCurrentLightSensorRate * 1000, mHandler);
+            mSensorManager.registerListener(mBackLightSensorListener, mBackLightSensor,
+                    mCurrentLightSensorRate * 1000, mHandler);
         } else if (!mAutoBrightnessOneShot && autoBrightnessOneShot) {
             mSensorManager.unregisterListener(mLightSensorListener);
+            mSensorManager.unregisterListener(mBackLightSensorListener);
         }
         mAutoBrightnessOneShot = autoBrightnessOneShot;
     }
@@ -698,6 +709,8 @@ public class AutomaticBrightnessController {
                 registerForegroundAppUpdater();
                 mSensorManager.registerListener(mLightSensorListener, mLightSensor,
                         mCurrentLightSensorRate * 1000, mHandler);
+                mSensorManager.registerListener(mBackLightSensorListener, mBackLightSensor,
+                        mCurrentLightSensorRate * 1000, mHandler);
                 return true;
             }
         } else if (mLightSensorEnabled) {
@@ -715,6 +728,7 @@ public class AutomaticBrightnessController {
             mHandler.removeMessages(MSG_UPDATE_AMBIENT_LUX);
             unregisterForegroundAppUpdater();
             mSensorManager.unregisterListener(mLightSensorListener);
+            mSensorManager.unregisterListener(mBackLightSensorListener);
         }
         return false;
     }
@@ -750,8 +764,11 @@ public class AutomaticBrightnessController {
             }
             mCurrentLightSensorRate = lightSensorRate;
             mSensorManager.unregisterListener(mLightSensorListener);
+            mSensorManager.unregisterListener(mBackLightSensorListener);
             mSensorManager.registerListener(mLightSensorListener, mLightSensor,
                     lightSensorRate * 1000, mHandler);
+            mSensorManager.registerListener(mBackLightSensorListener, mBackLightSensor,
+                    lightSensorRate * 1000, mHandler);
         }
     }
 
@@ -1027,6 +1044,7 @@ public class AutomaticBrightnessController {
         }
         if (mAutoBrightnessOneShot) {
             mSensorManager.unregisterListener(mLightSensorListener);
+            mSensorManager.unregisterListener(mBackLightSensorListener);
         }
     }
 
@@ -1395,8 +1413,24 @@ public class AutomaticBrightnessController {
                 // The time received from the sensor is in nano seconds, hence changing it to ms
                 final long time = (mDisplayManagerFlags.offloadControlsDozeAutoBrightness())
                         ? TimeUnit.NANOSECONDS.toMillis(event.timestamp) : mClock.uptimeMillis();
-                final float lux = event.values[0];
-                handleLightSensorEvent(time, lux);
+                mLux = event.values[0];
+                handleLightSensorEvent(time, Math.max(mLux, mBackLux));
+            }
+        }
+
+        @Override
+        public void onAccuracyChanged(Sensor sensor, int accuracy) {
+            // Not used.
+        }
+    };
+
+    private final SensorEventListener mBackLightSensorListener = new SensorEventListener() {
+        @Override
+        public void onSensorChanged(SensorEvent event) {
+            if (mLightSensorEnabled) {
+                final long time = mClock.uptimeMillis();
+                mBackLux = event.values[0];
+                handleLightSensorEvent(time, Math.max(mLux, mBackLux));
             }
         }
 
@@ -1633,4 +1667,9 @@ public class AutomaticBrightnessController {
             return new RealClock(offloadControlsDozeBrightness);
         }
     }
+
+    private Sensor getBackLightSensor() {
+        return SensorUtils.findSensor(mSensorManager, "xiaomi.sensor.back_lux", null,
+            SensorUtils.NO_FALLBACK);
+    }
 }
-- 
2.43.0

