From 47d2eec689b3ded87d9825a2e1602c9673f4c30f Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Thu, 28 Nov 2024 22:30:26 +0800
Subject: [PATCH] ignore uevent with null name

---
 core/java/android/os/UEventObserver.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/java/android/os/UEventObserver.java b/core/java/android/os/UEventObserver.java
index fa30e509f..e65ef2131 100644
--- a/core/java/android/os/UEventObserver.java
+++ b/core/java/android/os/UEventObserver.java
@@ -211,6 +211,9 @@ public abstract class UEventObserver {
 
             if (!mTempObserversToSignal.isEmpty()) {
                 final UEvent event = new UEvent(message);
+                   if(event.get("NAME")==null) {
+                    return;
+                }
                 final int N = mTempObserversToSignal.size();
                 for (int i = 0; i < N; i++) {
                     final UEventObserver observer = mTempObserversToSignal.get(i);
-- 
2.39.2

